packages/toolboxes/substrate/src/toolbox/baseSubstrateToolbox.ts

```diff
@@ -63,33 +67,48 @@ const createTransfer = (
 
 const transfer = async (
   api: ApiPromise,
-  signer: IKeyringPair,
+  signer: IKeyringPair | Signer,
   { recipient, assetValue, from }: SubstrateTransferParams,
 ) => {
   const transfer = createTransfer(api, {
     recipient,
     amount: assetValue.getBaseValue("number"),
   });
 
-  const tx = await transfer?.signAndSend(signer, {
-    nonce: await getNonce(api, from || signer.address),
+  if (isKeyringPair(signer)) {
+    const tx = await transfer?.signAndSend(signer, {
+      nonce: await getNonce(api, from || signer.address),
+    });
+    return tx?.toString();
+  }
+  const tx = await transfer?.signAndSend(from, {
+    signer,
+    nonce: await getNonce(api, from),
   });
```

Comment:
Early `return` statement outside of `guard clause` makes code less clear on what and when it happens - this one `tx` statement could be simple ternary and would resolve as it is over next lines - this makes code duplicated.

Reply:
fixed